src/main/java/org/opensearch/knn/index/memory/NativeMemoryAllocation.java

```diff
@@ -80,6 +81,26 @@ public interface NativeMemoryAllocation {
      */
     int getSizeInKB();
 
+    /**
+     * Increments the refCount of this instance.
+     *
+     * @see #decRef
+     * @throws IllegalStateException iff the reference counter can not be incremented.
+     */
+    default void incRef() {}
+
+    /**
+     * Decreases the refCount of this  instance. If the refCount drops to 0, then this
+     * instance is considered as closed and should not be used anymore.
+     *
+     * @see #incRef
+     *
+     * @return returns {@code true} if the ref count dropped to 0 as a result of calling this method
+     */
+    default boolean decRef() {
+        return true;
+    }
```

Comment:
> and should not be used anymore.

what is the meaning of this? Are you suggesting that is ref is not available for eviction?

Reply:
this comes from `org.opensearch.common.util.concurrent.RefCounted#decRef` and in `NativeMemoryAllocation.IndexAllocation.RefCountedReleasable#decRef` it would calling `closeInternal` when `count == 0`.  so this means: when count drops to 0, this allocation can not use anymore.